jgrapht-core/src/main/java/org/jgrapht/generate/PruferTreeGenerator.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * (C) Copyright 2018-2018, by Alexandru Valeanu and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.generate;
+
+import org.jgrapht.Graph;
+import org.jgrapht.GraphTests;
+
+import java.util.*;
+
+/**
+ * Generates a random tree using Prüfer sequences.
```

Comment:
As an alternative to random generation, would it make sense to also allow the caller to supply the sequence explicitly?

Reply:
This is a great idea. Thank you the suggestion.